tests/BridgeImplementationTest.php

```diff
@@ -1,188 +1,204 @@
-
-		require_once('bridges/' . $bridgeName . '.php');
-
-		$this->CheckBridgePublicFunctions($bridgeName);
-		$this->CheckBridgePublicConstants($bridgeName);
-		$this->CheckBridgeGetNameDefaultValue($bridgeName);
-		$this->CheckBridgeGetURIDefaultValue($bridgeName);
-
-	}
-
-	public function count() {
-		return count(Bridge::getBridgeNames());
+		$this->assertTrue(true);
 	}
 
-	public function run(TestResult $result = null) {
-
-		if ($result === null) {
-			$result = new TestResult;
+	/**
+	 * @dataProvider dataBridgesProvider
+	 */
+	public function testVisibleMethods($path) {
+		$allowedBridgeAbstract = get_class_methods('BridgeAbstract');
+		sort($allowedBridgeAbstract);
+		$allowedFeedExpander = get_class_methods('FeedExpander');
+		sort($allowedFeedExpander);
+
+		$this->setBridge($path);
+
+		$methods = get_class_methods($this->obj);
+		sort($methods);
+		if ($this->obj instanceof FeedExpander) {
+			$this->assertEquals($allowedFeedExpander, $methods);
+		} else {
+			$this->assertEquals($allowedBridgeAbstract, $methods);
 		}
+	}
 
-		foreach (Bridge::getBridgeNames() as $bridge) {
-
-			$bridge .= 'Bridge';
+	/**
+	 * @dataProvider dataBridgesProvider
+	 */
+	public function testMethodValues($path) {
+		$this->setBridge($path);
 
-			$result->startTest($this);
-			PHP_Timer::start();
-			$stopTime = null;
+		$value = $this->obj->getDescription();
+		$this->assertInternalType('string', $value, '$class->getDescription()');
+		$this->assertNotEmpty($value, '$class->getDescription()');
 
-			try {
-				$this->testBridgeImplementation($bridge);
-			} catch (AssertionFailedError $e) {
+		$value = $this->obj->getMaintainer();
+		$this->assertInternalType('string', $value, '$class->getMaintainer()');
+		$this->assertNotEmpty($value, '$class->getMaintainer()');
 
-				$stopTime = PHP_Timer::stop();
-				$result->addFailure($this, $e, $stopTime);
+		$value = $this->obj->getName();
+		$this->assertInternalType('string', $value, '$class->getName()');
+		$this->assertNotEmpty($value, '$class->getName()');
 
-			} catch (Exception $e) {
+		$value = $this->obj->getURI();
+		$this->assertInternalType('string', $value, '$class->getURI()');
+		$this->assertNotEmpty($value, '$class->getURI()');
 
-				$stopTime = PHP_Timer::stop();
-				$result->addError($this, $e, $stopTime);
+		$value = $this->obj->getIcon();
+		$this->assertInternalType('string', $value, '$class->getIcon()');
+	}
 
-			}
+	/**
+	 * @dataProvider dataBridgesProvider
+	 */
+	public function testUri($path) {
+		$this->setBridge($path);
 
-			if ($stopTime === null) {
-				$stopTime = PHP_Timer::stop();
-			}
+		$this->checkUrl($this->obj::URI);
+		$this->checkUrl($this->obj->getURI());
+	}
 
-			$result->endTest($this, $stopTime);
+	////////////////////////////////////////////////////////////////////////////
 
+	public function dataBridgesProvider() {
+		$bridges = array();
+		foreach (glob(self::PATH_BRIDGES . '*Bridge.php') as $path) {
```

Comment:
All PHP files under `bridges` must be bridges, so test all available PHP files.

```suggestion
		foreach (glob(self::PATH_BRIDGES . '*.php') as $path) {
```